Add support for targeting a build stage when building a Docker image

ClaytonNorthey92 opened the pull request:
allowing a developer the ability to choose a build stage using the `withTarget` method, see [discussion here](https://github.com/testcontainers/testcontainers-node/discussions/619)

fixes [629](https://github.com/testcontainers/testcontainers-node/issues/629)

cristianrgreco commented on src/testcontainers/src/generic-container/generic-container-builder.ts:
```diff
@@ -16,6 +16,7 @@ export class GenericContainerBuilder {
   private buildArgs: BuildArgs = {};
   private pullPolicy: ImagePullPolicy = PullPolicy.defaultPolicy();
   private cache = true;
+  private target?: string = undefined;
```
```suggestion
  private target?: string;
```

cristianrgreco commented on src/testcontainers/src/generic-container/generic-container.test.ts:
```diff
@@ -434,4 +435,39 @@ describe("GenericContainer", () => {
+
+    const logLines: Array<string> = [];
+
+    const recordLogs = async (name: string, container: StartedTestContainer) => {
```
Is there an easier way to tell the difference between both containers, e.g could each set a label?

cristianrgreco commented on src/testcontainers/src/generic-container/generic-container.test.ts:
```diff
@@ -434,4 +435,39 @@ describe("GenericContainer", () => {
+
+  it("should build a target stage", async () => {
+    const context = path.resolve(fixtures, "docker-multi-stage");
+    const firstConatiner = await GenericContainer.fromDockerfile(context).withTarget("first").build();
```
Typo `firstConatiner` ->  `firstContainer`

cristianrgreco commented on src/testcontainers/src/generic-container/generic-container.test.ts:
```diff
@@ -434,4 +435,39 @@ describe("GenericContainer", () => {
+
+  // failing to build an image hangs within the DockerImageClient.build method,
+  // that change might be larger so leave it out of this commit but skip the failing test
+  it.skip("should throw an error for a target stage that does not exist", async () => {
```
Good idea, thanks, I can look into this at some point

ClaytonNorthey92 commented:
hey @cristianrgreco thanks for the feedback, I...
* removed the `= undefined` from the `target` attribute definition in `generic-container-builder.ts`
* using a label `stage` when testing, opposed to logs, to simplify the test
* fixed the type you pointed out
* updated the `README`

let me know if there is anything else 👍
